crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -233,47 +231,37 @@ async fn monitor_tedge_service(
         }
 
         let elapsed = start.elapsed();
-        if elapsed < tokio::time::Duration::from_secs(interval) {
-            tokio::time::sleep(tokio::time::Duration::from_secs(interval) - elapsed).await;
+        if elapsed < Duration::from_secs(interval) {
+            tokio::time::sleep(Duration::from_secs(interval) - elapsed).await;
             warn!("tedge systemd watchdog not started because no services to monitor")
         }
     }
 }
 
 async fn get_latest_health_status_message(
-    request_timestamp: String,
+    request_timestamp: OffsetDateTime,
     messages: &mut mpsc::UnboundedReceiver<Message>,
 ) -> Result<HealthStatus, WatchdogError> {
-    loop {
-        if let Some(message) = messages.next().await {
-            if let Ok(message) = message.payload_str() {
-                debug!("Health response received: {}", message);
-                if let Ok(health_status) = serde_json::from_str::<HealthStatus>(message) {
-                    let request_timestamp = OffsetDateTime::parse(
-                        &request_timestamp,
-                        &format_description::well_known::Rfc3339,
-                    )?
-                    .unix_timestamp();
-                    let datetime = OffsetDateTime::parse(
-                        &health_status.time,
-                        &format_description::well_known::Rfc3339,
-                    )?
-                    .unix_timestamp();
-
-                    if datetime >= request_timestamp {
-                        return Ok(health_status);
-                    } else {
-                        debug!(
-                            "Ignoring stale health response: {:?} older than request time: {}",
-                            health_status, request_timestamp
-                        );
-                    }
+    while let Some(message) = messages.next().await {
```

Comment:
I figured "loop indefinitely if the channel closes" is likely not the desired behaviour, and could plausibly lead to the actor not shutting down, so I've change the function to return an error when we stop receiving messages.